src/server/test/util/vaidationHelpers.js

```diff
@@ -0,0 +1,145 @@
+	const token = await getToken(); 
+	const payload = { ...basePayload };
+	if (invalidValue === undefined) {
+		delete payload[field];
+	} else {
+		payload[field] = invalidValue;
+	}
+	const res = await chai.request(app)
+		.post(endpoint)
+		.set('token', token) 
+		.send(payload);
+
+	expect(res).to.have.status(400);
+}
+
+/**
+ * Validates that the API rejects payloads where minVal is greater than maxVal.
+ *
+ * @param endpoint the API endpoint URL to test (e.g., /api/units/addUnit)
+ * @param basePayload a valid payload object to use as the base for testing
+ */
+async function validateMinMaxRelation({ endpoint, basePayload }) {
+	const token = await getToken(); 
+    const invalidPayload = {
+        ...basePayload,
+        minVal: (basePayload.minVal || 10) + 1,  
+        maxVal: basePayload.minVal || 10 
+    };
+    const res = await chai.request(app)
+        .post(endpoint)
+		.set('token', token) 
+        .send(invalidPayload);
+    
+    expect(res).to.have.status(400);
+}
+
+/**
+ * Validates a string field by testing required presence, min/max length, and enum constraints.
+ *
+ * @param field the name of the string field to validate
+ * @param endpoint the API endpoint to test (e.g., /api/units/addUnit)
+ * @param basePayload a valid payload object to start from
+ * @param required whether the field is required (default: true)
+ * @param minLength the minimum length allowed for the string (default: 1)
+ * @param maxLength the maximum length allowed for the string (default: 255)
+ * @param enumValues optional array of valid enum values to test against
+ */
+async function validateString({ field, endpoint, basePayload, required = true, minLength = 1, maxLength = 255, enumValues = null }) {
+
+	if (required) {
+		await testInvalidField({ field, invalidValue: undefined, endpoint, basePayload });
+	}
+
+	if (minLength > 0) {
+		await testInvalidField({ field, invalidValue: 'x'.repeat(minLength - 1), endpoint, basePayload });
+	}
+
+	await testInvalidField({ field, invalidValue: 'x'.repeat(maxLength + 1), endpoint, basePayload });
+
+	if (enumValues) {
+		await testInvalidField({ field, invalidValue: 'INVALID_ENUM', endpoint, basePayload });
+	}
+}
+
+/**
+ * Validates an integer field by testing for presence (if required), 
+ * numeric bounds (min and max), and type correctness.
+ *
+ * @param field the name of the integer field to validate
+ * @param endpoint the API endpoint to test (e.g., /api/units/addUnit)
+ * @param basePayload a valid base object used to construct requests
+ * @param required whether the field is required (default: true)
+ * @param min the minimum allowed integer value (optional)
+ * @param max the maximum allowed integer value (optional)
+ */
+async function validateInt({ field, endpoint, basePayload, required = true, min = null, max = null }) {
+	if (required) {
+		await testInvalidField({ field, invalidValue: undefined, endpoint, basePayload });
```

Comment:
If I understand the code, the test is sending the field in the route with undefined. This seems different than it being absent when required.